lib/streamlit/DeltaGenerator.py

```diff
@@ -198,6 +198,17 @@ def __init__(
         self._container = container
         self._path = path
 
+    def __getattr__(self, name):
+        def wrapper(*args, **kwargs):
+            if self._container == BlockPath_pb2.BlockPath.SIDEBAR:
+                raise RuntimeError("'%s' was called but "
+                                   "it's not available for sidebar" % name)
+            else:
+                raise RuntimeError("'%s' was called but "
+                                   "it's not an existing method" % name)
```

Comment:
It would be great if we could check whether `name` is a valid `st` method (from `__init__.py`), and then have different messages depending on the case:

* If `name` is a valid `st` method:
  * If container is `SIDEBAR`
    ```
    Method `%{name}s()` does not exist for `st.sidebar`. Did you mean `st.%{name}s()`?
    ```

  * else:
    ```
    Method `%{name}s()` does not exist for `DeltaGenerator` objects. Did you mean `st.%{name}s()`?
    ```

* else:
    ```
    `%{name}s()` is not a valid Streamlit command.
    ```